src/libraries/Microsoft.PowerFx.Core/Public/Types/FormulaType.cs

```diff
@@ -148,7 +151,7 @@ internal static FormulaType Build(DType type)
                     return BindingError;
 
                 default:
-                    throw new NotImplementedException($"Not implemented type: {type}");
+                    return new UnsupportedType(type);
```

Comment:
Don't know how sane this is, but it makes it really convenient for what we're trying to do.

Reply:
Note that this type does not have a public way of constructing it.